Add new license Anti-Capitalist Software License #2362

sritasngh opened the pull request:
<!-- Delete Template sections if unneccesary -->
<!-- Add issue number here (We encourage you to create the Issue First) -->
<!-- You can also link the issue in Commit Messages -->
Signed-off-by: itssingh saritasingh.0425@gmail.com
Fixes #2362


<!--
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!
* [x] - Checked Box
* [ ] - Unchecked Box
-->
 Added new licenses ANTI-CAPITALIST SOFTWARE LICENSE v-1.0, v-1.1, v-1.2, v-1.3 and v-1.4
### These added licenses are giving correct scan result. 

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

<!--
We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged**
if your tests fail at first!
If tests do fail, click on the red `X` to learn why by reading the logs.
Thanks!
-->

<!-- Don't forget to Signoff -->

JonoYang committed:
Process deps from dependencyManagement #2103

Signed-off-by: Philippe Ombredanne <pomnbredanne@nexb.com>

mrmayurgithub committed:
Add basic support Dart pub packages #2110

Signed-off-by: Mayur Agarwal <mr.mayurrrr@gmail.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>
Fix import #2110

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new generated keywords

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

abhi27-web committed:
Merge pull request #2375 from abhi27-web/patch-1

Update CONTRIBUTING.rst

Authored-by: Abhigya Verma<abhigyaverma27@gmail.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.0.LICENSE:
```diff
@@ -0,0 +1,14 @@
+Copyright © [year] [copyright holders]
```
Please do not include a "dummy" copyright template in the license text 
```suggestion
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.0.yml:
```diff
@@ -0,0 +1,10 @@
+text_urls: 
+    - https://anticapitalist.software/past_versions/v1-0.html
+ignorable_urls: 
+    - https://anticapitalist.software/past_versions/
```
Can you add a trailing line return?

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.1.LICENSE:
```diff
@@ -0,0 +1,14 @@
+Copyright © [year] [copyright holders]
```
Same above: no copyright template there
```suggestion
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.2.LICENSE:
```diff
@@ -0,0 +1,15 @@
+Copyright © [year] [copyright holders]
```
Same as above.
```suggestion
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.3.LICENSE:
```diff
@@ -0,0 +1,20 @@
+ANTI-CAPITALIST SOFTWARE LICENSE (v 1.3)
+
+Copyright © [year] [copyright holders]
```
```suggestion
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.4.LICENSE:
```diff
@@ -0,0 +1,21 @@
+ANTI-CAPITALIST SOFTWARE LICENSE (v 1.4)
+
+Copyright © [year] [copyright holders]
```
```suggestion
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.3.yml:
```diff
@@ -0,0 +1,10 @@
+text_urls: 
+    - https://anticapitalist.software/past_versions/v1-3.html
+ignorable_urls: 
+    - https://anticapitalist.software/past_versions/
```
Add trailing LF

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.3.LICENSE:
```diff
@@ -0,0 +1,20 @@
+
+4. If the User is an organization, then the User is not law enforcement or military, or working under either.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT EXPRESS OR IMPLIED WARRANTY OF ANY KIND, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE ...
```
Add trailing LF

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.2.yml:
```diff
@@ -0,0 +1,10 @@
+text_urls: 
+    - https://anticapitalist.software/past_versions/v1-2.html
+ignorable_urls: 
+    - https://anticapitalist.software/past_versions/
```
Add trailing LF

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.2.LICENSE:
Add trailing LF

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.1.yml:
```diff
@@ -0,0 +1,10 @@
+text_urls: 
+    - https://anticapitalist.software/past_versions/v1-1.html
+ignorable_urls: 
+    - https://anticapitalist.software/past_versions/
```
Add trailing LF

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.1.LICENSE:
Add trailing LF

pombredanne reviewed (changes requested):
Thanks! I have a few nitpickings for your consideration before we can merge.

pombredanne reviewed (changes requested):
@itssingh  Thank you! Almost there too here!

See the "ignorables"-related test failures in https://dev.azure.com/nexB/scancode-toolkit/_build/results?buildId=3262&view=logs&j=3c16b29b-eb82-5c7a-26ca-153f5e2ef8d4&t=525ae439-c19a-5d45-abf1-df81b3df2f93&l=43551 for your review.

sritasngh commented:
> See the "ignorables"-related test failures

should I put `url` in `notes:` ?

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.4.yml:
```diff
@@ -0,0 +1,9 @@
+category: Free Restricted
+owner: Everest Pipkin and Ramsey Nasser
+homepage_url: https://anticapitalist.software/
+ignorable_urls: 
```
```suggestion
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.4.yml:
```diff
@@ -0,0 +1,9 @@
+owner: Everest Pipkin and Ramsey Nasser
+homepage_url: https://anticapitalist.software/
+ignorable_urls: 
+    - https://anticapitalist.software/past_versions/
```
```suggestion
```

pombredanne commented on src/licensedcode/data/rules/anti_capitalist1.yml:
```diff
@@ -0,0 +1,5 @@
+license_expression: anti-capitalist-1.4
+is_license_text: yes
+ignorable_urls: 
+    - https://anticapitalist.software/past_versions/
```
```suggestion
```

pombredanne commented on src/licensedcode/data/rules/anti_capitalist1.yml:
```diff
@@ -0,0 +1,5 @@
+license_expression: anti-capitalist-1.4
+is_license_text: yes
+ignorable_urls: 
```
```suggestion
```

pombredanne reviewed (commented):
> should I put url in notes: 

no, you just need to add `ignorable_*` fields for URLs and emails

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.4.yml:
```diff
@@ -0,0 +1,8 @@
+category: Free Restricted
+owner: Everest Pipkin and Ramsey Nasser
+homepage_url: https://anticapitalist.software/
+
```
```suggestion
spdx_license_key: LicenseRef-scancode-anti-capitalist-1.4
```

pombredanne commented on src/licensedcode/data/licenses/anti-capitalist-1.4.yml:
```diff
@@ -0,0 +1,8 @@
+owner: Everest Pipkin and Ramsey Nasser
+homepage_url: https://anticapitalist.software/
+
+    
```
```suggestion
```

pombredanne commented on src/licensedcode/data/rules/anti_capitalist1.yml:
```diff
@@ -0,0 +1,3 @@
+license_expression: anti-capitalist-1.4
+is_license_text: yes
+    
```
```suggestion
```

pombredanne commented on src/licensedcode/data/rules/anti_capitalist2.yml:
```diff
@@ -0,0 +1,3 @@
+license_expression: anti-capitalist-1.4
+is_license_text: yes
+    
```
```suggestion
```

pombredanne commented on src/licensedcode/data/rules/anti_capitalist3.yml:
```diff
@@ -0,0 +1,3 @@
+license_expression: anti-capitalist-1.4
+is_license_text: yes
+    
```
```suggestion
```

pombredanne commented on src/licensedcode/data/rules/anti_capitalist4.yml:
```diff
@@ -0,0 +1,3 @@
+license_expression: anti-capitalist-1.4
+is_license_text: yes
+    
```
```suggestion
```

pombredanne reviewed (changes requested):
Thank you! Please consider the few cleanups I suggested. Also feel free to squash all your commits in a single one.

sritasngh commented:
> consider the few cleanups I suggested

I left these https://github.com/nexB/scancode-toolkit/pull/2364#discussion_r644868769 as trailing line return        https://github.com/nexB/scancode-toolkit/pull/2364#discussion_r583924347 as you previously suggested. Removed now :)

pombredanne committed:
Add new misc. generated code clues

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

sritasngh committed:
Add emails and urls

Add emails and urls in converted_infos. Add holders, authors, emails and
urls information in html template.

Signed-off-by: Sarita Singh <saritasingh.0425@gmail.com>
Co-authored-by: Avinal Kumar <avinal.xlvii@gmail.com>

sritasngh committed:
Add test for sample HTML template and update changelog

Signed-off-by: Sarita Singh <saritasingh.0425@gmail.com>

Co-authored-by: Avinal Kumar <avinal.xlvii@gmail.com>

pombredanne committed:
Bump commoncode

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

AyanSinhaMahapatra committed:
Add rules to improve debian-slim license detection

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Regenerate tests after adding rules

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add filter for license intros to reduce unknowns

Adds a filter for unstructured debian copyright file, where if
license intros are perfectly matched, they are discarded, as
in the context of a debian copyright file, the license texts/notices
are also there and not just the intro.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add tests to check for unknowns and issues

Adds tests which fail if there is a unknown license
detection or a license detection issue with low match
coverage present in the test cases. Also traces the
detections in case of failures. Fixes some text expectations.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne committed:
Merge pull request #2557 from nexB/bump-commoncode

Bump commoncode

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into alpianon-patch-documentnamespace

pombredanne committed:
Add CHANGELOG entry for SPDX format update

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Streamline SPDX formatting code #2344

Format code. Refactor common code between RDF and TV.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update SPDX test expectatations #2344

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use correct pytest setting for markers

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Clarify function to extract data from wheel name

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'pip2pi/master' into improve-thirdparty-utilities

akugarg committed:
Add is_unknown flag for license class

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Add is_unknown flag to data files of unknown licenses

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

pombredanne committed:
Ignore UUIDs in SPDX namespeca tests #2344

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not import deprecated debut module

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not use deprecated execute2 function

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not use deprecated Type.object in jsonstreams

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2351 from alpianon/alpianon-patch-documentnamespace

added DocumentNamespace tag

pombredanne committed:
Update expected results for generated code

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Also include setuptool_scm in dependencies

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2563 from nexB/remove-deprecation-warnings

Remove deprecation warnings

pombredanne committed:
Merge pull request #2564 from nexB/improve-thirdparty-utilities

Improve thirdparty utilities

JonoYang committed:
Add code to parse MSI files for metadata #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Use new collections.abc for Set

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Save all MSI data to extra_data #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Run msiinfo with commoncode.command.execute #2565

    * Create get_msiinfo_bin_location()

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Add test for MSI parser #2559

Signed-off-by: Jono Yang <jyang@nexb.com>

akugarg committed:
Merge branch 'nexB:develop' into add_new_flag

akugarg committed:
Add is_unknown flag to data files of unknown licenses

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Update test cases for new flag addition

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

ldrumm committed:
Add Meson build language support (#1823)

* Add a Meson language lexer

* update the mappings

* Add meson to the list of supported languages

* Add a meson.build example file

sblondon committed:
Add procfile lexer (#1808)

* add Procfile Lexer

* add unit tests for ProcfileLexer

* add Procfile lexer author

* Document Procfile as supported language

* add Procfile.output file

* merge Profile files; update output file

* Add versionadded info

* Fix typo (processus -> processes)

* detect Integer only

* split Text from Whitespace

pombredanne committed:
Bump extractcode

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

akugarg committed:
Update test cases for new flag addition

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Add new validation rules

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

AyanSinhaMahapatra committed:
Add and modify rules for debian-slim copyrights

Add new rules and modify existing rules to get debian-slim license
detections correct.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify test expectations for debian-slim copyright

Modify test expectations after license detection improvements in debian-slim
copyright files.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Improve debian copyright and license detection

Improves license detection by modifying unstructured license
intro detection with full coverage. Fixes matched_text bug in unknown
debian license by setting the lines. Removes unknown copyright detections.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

JonoYang committed:
Hook up MSI parsing #2559

Signed-off-by: Jono Yang <jyang@nexb.com>

AyanSinhaMahapatra committed:
Improve tests tracing for unknown matches

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add license_matches property for strucutred files

Adds license_matches property to get the LicenseMatch objects
out of LicenseDetection objects directly as a property.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

JonoYang committed:
Report installed Windows programs #2565

    * Fix function that reports installed .NET versions

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Add Package for installed Windows programs #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Add tests for registry parsing #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

akugarg committed:
Update test cases for new flag addition

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

JonoYang committed:
Create functions for use in scancode.io pipeline

    * Collect installed file locations

Signed-off-by: Jono Yang <jyang@nexb.com>

AyanSinhaMahapatra committed:
Detect structured debian copyright files better

Structured debian copyright files are deteected by the 'format: '
first line, and this adds more format links commonly encountered in
debian copyright files, thus detecting structured copyright files
better.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

JonoYang committed:
Check multiple locations for registry files #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

AyanSinhaMahapatra committed:
Add rules to remove unknowns in debian copyrights

Adds rules to remove unknown-license-references in common debian
copyright files. Regenerates test files with removed unknowns.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Bump license-expression to latest 21.6.14 version

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Use licensing.dedup to simplify expressions

Use the licensing.dedup function from license-expression library
to simplify the licenses without losing any license-expression
specific information.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Regenerate tests with deduped license-expressions

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

JonoYang committed:
Catch NoRegistrySubkeysException #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

AyanSinhaMahapatra committed:
Add primary license for structured debian files

For structured debian copyright files, return an attribute
with the primary license detected, and for unstructured files
return None.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Regenerate tests with primary license included

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Reproduce bug in get_license_expression

Adds the debian copyright file which caused an exception, as
it says it's a structured debian copyright file, but doesn't
have structured paragraphs in it.

See - https://github.com/nexB/scancode.io/issues/219

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Fix bug in get_license_expression

Fixes bug in get_license_expression by not adding a LicenseDetection
object when there are no license matches in an other paragraph,
in a structured debian copyright file.

See - https://github.com/nexB/scancode.io/issues/219

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify processing None at license_expressions

Modify get_license_expression to not pass None values to
combine_expression, and also handle if all license detections
are None, by getting a expression from the license_matches,
or raising an Error if no license_matches.

See - https://github.com/nexB/scancode.io/issues/219

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

akugarg committed:
Fix failing test cases

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

AyanSinhaMahapatra committed:
Add more consistency checks and fix test failures

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne committed:
Reorganize code to match scancode target way

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove unused lexers

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format and streamline code and headers

Add origin docs.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove unused code. Make tests pass

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'local/master' into develop

AyanSinhaMahapatra committed:
Modify filter_licenses for license expressions

Remove filter_licenses flag from get_license_expression functions
as licensing.dedup() makes this function redundant. Also renames
filter_licenses to filter_duplicates.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne committed:
Merge pull request #2570 from nexB/update-deps-and-packaging

Bump extractcode

pombredanne committed:
Merge pull request #2572 from nexB/bump-commoncode

Also include setuptool_scm in dependencies

pombredanne committed:
Merge pull request #2573 from nexB/remove-deprecation-warnings

Use new collections.abc for Set

AyanSinhaMahapatra committed:
Remove "none" copyrights

Remove copyrights which starts with "none", like
`Copyright: none`. Regenerate tests to remove nones
from results.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Update docstrings and functions

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify reporting unknown matches

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Move consistency errors function into EDC

Move consistency error function into the EnhancedDebianCopyright class
and have the error textx added directly instead of a dict lookup.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Rename rules to have appropriate names

Rename the rules appropriately, with some modifications and additions.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add check for debian expression substitution

Adds a function to check whether the keys in debian expression can be substituted
successfully, and adds an UnknownMatch if there are inconsistencies.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add license detections from extra_data

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Fix subs bug for inconsistent cases

Adds a function to do license detection on the license name
if it is not in the seen keys. Regenerates test and adds rules.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Remove old debian_copyright parsing functions

Removes old debian copyright parsing functions and also
removes debian_licenses.txt.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Fix failing tests

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne committed:
Use line_number as a var name

This is more explicit and readable

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2558 from nexB/2390-improve-debian-license-detection

Improve debian license detection #2390

pombredanne committed:
Adopt new simplified Pygmars with line tracking

We now have correct license numbers returned by copyright and authors
detection. Also fix some minor copyright detection issues.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Split windows_helper.py into two files #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

akugarg committed:
Merge branch 'nexB:develop' into add_new_flag

pombredanne committed:
Use pygmars >= 0.6.b1

This is required for the new, better and faster copyright detection

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Handle some multiline copyrights #1565

If there is only a lien trailing with a year, AND or BY we consider
the next empty line as part of the copyright statement.

Also treat some camel case as JUNK and Module as NN

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into copyrights-improvements

JonoYang committed:
Refactor report_installed_programs #2565

    * Create get_registry_tree that returns a list of dicts that contain the registry entries
    * report_dotnet_versions and report_installed_programs is now broken up into two parts, first is a call it get_registry_tree and a second is a call to _report_installed_programs or _report_dotnet_versions. These functions prefixed with underscores takes in a list of dictionaries
    * Update tests by replacing the SOFTWARE registry file with json results

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Make functions more conducive to testing #2559

    * Remove test MSI installer

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Report other known program files #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

akugarg committed:
Add is_unknown flag in data model for unknown licenses

This reverts commit 4fa04ffbef16ec944d2c0b42f32f2313b0b88b11.

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

JonoYang committed:
Add and update win_reg tests #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Add test for get_installed_files #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Improve test fixture regeneration

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update ignorables with latest copyright detection

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Factor common license ignorables function

Also used in tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove trailing LF

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve copyright detection

- Allow multiline copyright detection with an empty line if ending with
  a conjunction or a year.
- Do not treat Free as an NNP
- Better handle leading and trailing comment line markers
- Always enable tracing with deep tracing

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Parse Microsoft Update Manifest files #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
GUID is not a word in a copytights

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Small is not a proper noun

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not match short rules with stopwords laced in

No license should be detected in "License a Private"
with a rule "license Private" as there is a stopword inside and
this is a short rule.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use bsd-new is rather bsd-simplified

This notice is vague and matches references to bsd-new or
bsd-simplified when seen in the wild. Therefore we lower the relevance
and reference bsd-new which covers also bsd-simplified.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Boost rule relevance

This is unambiguously a CC0 license reference

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format docstring

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Bump pygmars

Also align setup-mini.cfg with setup.cfg

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add updated Google proprietary licenses

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2590 from nexB/copyrights-improvements

Improve copyrights detection

akugarg committed:
Merge branch 'develop' into add_new_flag

pombredanne committed:
Merge pull request #2538 from nexB/new-generated-code-clues

Add new generated code clues

pombredanne committed:
Improve handing of pub deps

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add notes to RULE to avoid confusion


Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new generic way to tests package parsers

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add pubspec parsing tests #2110

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Collect pubspec deps, authors and extra data #2110

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Treat pinned maven versions as resolved

Also update tests with dependencyManagement collections and
treat compile-time dependencies not as runtime.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Enable pub detection in scans

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add support for pub lockfiles #2110

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Add regipy and packagedcode-msitools as deps #2565

    * Add cli tests for win_pe and mum package detection
    * Update test results

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into shell-parser

pombredanne committed:
Merge pull request #2134 from nexB/2103-collect-deps-from-dependencymanagement

 Process deps from dependencyManagement #2103

pombredanne committed:
Merge pull request #2571 from nexB/2110-dart-pubspec

Add basic support Dart pub packages #2110

pombredanne committed:
Rename test directory

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add simple shell parser

This can parse shell scripts to extract and expand variables.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into shell-parser

pombredanne committed:
Add more bash parsing tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove dangling print statement

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Create function that checks bazel rule names #2594

    * Update test data to ensure bazel parser only creates packages for rules that end with `binary` or `library`

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Set regen=False for package cli tests

Signed-off-by: Jono Yang <jyang@nexb.com>

akugarg committed:
Update test cases upto expectations

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

pombredanne committed:
Make shell parsing work

Improve tests.
Use latest pygmars
Use newer parameter-expansion2 (temp advanced patched version)

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Update test results #2565

    * Output installed_files in test result files

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Use common datra driven utility function

And more and regenerate lexing tests using this new function

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2595 from nexB/2594-report-proper-bazel-rules

2594 report proper bazel rules

JonoYang committed:
Use relative file paths for installed_files #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Merge branch 'develop' into 2565-detect-installed-windows-programs

JonoYang committed:
Sort installed_files by path #2565

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Merge pull request #2582 from nexB/2565-detect-installed-windows-programs

Detect installed windows programs

JonoYang committed:
Bump regipy version to 2.0.0

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Bump regipy version to 2.0.0

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Improve shell parsing

- Use improved parameter-expansion
- Parse bash arrays
- Simplify lexing and grammar for better readability
- Add new "needed_variables" argument to parsing top-level function to
  limit the set of variables and errors returned to these named in this
  set.
- Improve tracing and variable resolution logic.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Check status.d dir if status file does not exist

    * Distroless images sometimes place status files for individual packages in the status.d dir instad of using a status file

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Collect empty bash variables

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Filter bash parsing on needed_variables

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

akugarg committed:
Add is_unknown flag in data model for unknown licenses

This reverts commit a1843cf307788a736e9bee9566a0909326249e57.

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

pombredanne committed:
Add new ttypo license

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Accept extra symbols for license expression

In license detection, we now accept extra expression symbols to use
for license expression detection. This enables the support of custom
SPDX-like license symbols used in some package repos and extra non-SPDX
symbols thta are also commonly used.

- Add a new function "match_spdx_lid.get_spdx_expression()"
- Add new argument "expression_symbols" to all needed functions and
methods.
- Update doc strings accordingly

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Parse .mui files using win_pe Package detection

    * Update tests

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Bump commoncode and extractcode

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Update misqualified license rule.

This rule is mostly bsd-new and seldom bsd-simplified. We now account
for this with reporting a bsd-new of gpl and a lowered relevance

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new OpenSSL AGPL exception license

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Correct OpenSSL exceptions to be Copyleft

These are exceptions to Copyleft licenses that strictly apply to
OpenSSL and do not turn the licenses in a some Copyleft Limited license.
Therefore I am making these Copyleft instead of Copyleft Limited.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve Alpine manifest handling

- Parse APKBUILDs including shell parameter expansion using a shell
  parser and some minimal fixes to ensure we can resolve all key
  package metadata. (in particular sources)
- Refine installed packages data processing to align this with APKBUILD
  parsing
- Add new declared_license detection that can correctly detect licenses
  on all Alpine 3.13 and 3.14 packages. It is composed of expression
  syntax fixes, flexible expression parsing and non-SPDX symbol
  substitution

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Correct test expectation

This was passing with previsou license updates

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into shell-parser

pombredanne committed:
Do not fail on empty expression

In the refactoring of 64ad11425295fe2296685557c6a67b867b31fc7b a test
was not carried over to ensure we can handle empty license expression
such as "SPDX-License-Identifier: " and not attempt to return an empty
match (with an empty Rule which would fail) if this happens.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refine Debian copyright handling

- Apply minor cleanups and formatting
- Refactor EnhancedDebianCopyright to use consistently the method
  debian_copyright.paragraphs() to select parapgraphs.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Bump virtualenv

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Parse .mun files in addition to .mui files #2600

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Parse .mun, .tlb, .winmd, and .com files #2600

    * Update tests

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Correct new docstring test

Use proper quotes rather than backticks

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Bump minimum coverage for some ambiguous rules

These rules were detected incorrectly and theyu are closely related to
other simpler rules. Bumping the coverage helped fix the issues

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update canary tokenizing test

This test nee to be updated sometimes when we add new words in the
license index. This is such a case

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Fix docstring typo

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update canary tokenizing test

This test nee to be updated sometimes when we add new words in the
license index. This is such a case. Added a comment to clarify why this
is happening.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format and clean sequence matching test

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use dictionaries when testing parsing

This is better than representation

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update deps

Bump pygmars
Remove setuptools, wheel and pip pins

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2598 from nexB/shell-parser

Add shell parser
Improve Alpine license detection
Add Alpine APKBUILD parser

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update changelog

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Remove is_debian_status_file

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Update packagedcode plugin CLI tests

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Align tests for pubspec with latest code

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2609 from nexB/bump-commoncode-and-extractcode-version

Bump commoncode and extractcode

pombredanne committed:
Merge pull request #2606 from nexB/parse-mui-files

Parse mui files

pombredanne committed:
Merge pull request #2608 from nexB/bump-regipy-version

Bump regipy version to 2.0.0

pombredanne committed:
Merge pull request #2607 from nexB/update-debian-package-getter

Update debian package getter

pombredanne committed:
Improve ignorable regeneration

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Set file paths relative to Docker layer root #2615

    * Update tests

Signed-off-by: Jono Yang <jyang@nexb.com>

JonoYang committed:
Add wildcard character after extensions #2615

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Requalify rule as gpl-2.0-plus

XVId is under the gpl-2.0-plus, not the gpl-2.0.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update CHANGELOG

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Bump version to 21.7.30

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve copyright detection

These are corner cases found in handling "ignorables" clues for
the latest licenses and license rules.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Fix typo in comment

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not filter_if_only_known_words_rule for small

"filter_if_only_known_words_rule()" is no longer apply for small
license detection rules. This was discarding too many matches. Instead
the correct approach is to create false positive detection rules for
the few cases that we encounter.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update license ignorable clues

Thse are now aligned with the latest copyright detection

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve existing rule ignorables

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Check for installed x86 programs

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Add copyleft license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new permissive license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new misc. license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Update test results #2615

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Cleanup aggressively older egg installation

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Cleanup egg-info in configure or release build

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Align tests expectation with latest license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not treat binary sociall for matched text

Return whole lines based on requested option, not based on filetype

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Align tests expectation with latest license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Add root_prefix arg to get_installed_packages #2614

    * The root prefix will be passed into get_installed_packages instead of assuming that the root_prefix is always "Files"
    * Update win_pe extensions
    * Update test files

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Align debian copyright tests with latest rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format code

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Test Debian copyright for full match details

The Debian copyright tests expected data were just a list of items
without an attribute name telling what each item was.

This fixes this by making the expected YAML results a dictionary rather
than a list. It also adds the full match details to the expected results such that we can better assess what failed and how and can better
update the tests as we improve license detection.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve scancode release creation

- Use correct py command for Windows (it was incorrect in 21.6.7 and
  21.7.31. This is fixing #2610
- Build and test apps for all supported Python versions

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Bump version to v21.8.1

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve documentation

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Lint and cleanup documentation

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

JonoYang committed:
Update expected test results #2615

Signed-off-by: Jono Yang <jyang@nexb.com>

pombredanne committed:
Bump to v21.8.4

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

akugarg committed:
Add new license for balloon open hardware

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Add new license for defensive patent license

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Add new license for Permissive binary license #2617

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

pombredanne committed:
Improve release creation speed

- Allow to work from a checkout of wheels
- Add new --continue option to restart archive creation where left

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2623 from nexB/2615-installedwindowsprogram-installed-files-paths

Collect InstalledWindowsProgram installed files #2615

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into add_new_flag

pombredanne committed:
Update changelog

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add comment, docstring and format code

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2627 from nexB/release_creation_speed

Improve release creation speed

pombredanne committed:
Merge pull request #2626 from nexB/omnibus-license-updates-july-21

Omnibus license updates July/Aug 21

pombredanne committed:
Merge pull request #2548 from akugarg/add_new_flag

Add new flag in License Data Model definition

akugarg committed:
return referenced_filenames for licenses matches

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

mapelpapel committed:
Update Contributing: Development setup-instructions

Some instructions were not (exactly) right, and some commands are now also highlighted properly.

Signed-off-by: Marcel Rettig <mapelpapel@gmail.com>

pombredanne committed:
Merge pull request #2631 from mapelpapel/update-contributing

Update Contributing: Development setup-instructions

Signed-off-by: Marcel Rettig <mapelpapel@gmail.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

akugarg committed:
Add new talis community license

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

sritasngh commented:
@pombredanne Should I add notes for all the rules?

akugarg committed:
Update test cases upto expectations

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

pombredanne committed:
Merge pull request #2632 from akugarg/update_api

Referenced_filenames should be returned by API function

petergardfjall committed:
adds MIT detection rule 1055

This provides a rule that matches MIT notices of form (note: indent for highlight):

  License

  This project is licensed under the terms of the MIT license.

Signed-off-by: Peter Gardfjäll <peter.gardfjall.work@gmail.com>

petergardfjall committed:
apache-2.0_328.RULE: set minimum_coverage to 70

This avoids Apache-2.0 "overriding" mit_349.RULE in cases like

  ## License

  This project is licensed under the terms of the MIT license.

Signed-off-by: Peter Gardfjäll <peter.gardfjall.work@gmail.com>

pombredanne committed:
Merge pull request #2539 from itssingh/html_temp

Add emails and urls to HTML output

pombredanne committed:
Merge pull request #2636 from petergardfjall/issue-2635

Avoid misinterpreting MIT license notice as Apache-2.0, issue #2635

akugarg committed:
Merge branch 'nexB:develop' into add_new_licenses

akugarg committed:
Update test cases upto expectations

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Add final report for GSoC'21

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>

akugarg committed:
Merge pull request #2648 from akugarg/final_report

Add final report for GSoC'21

yns88 committed:
Add "--no-check-version" CLI option to scancode

This version check causes problems when running in a firewalled environment (in my case, hanging for 30 seconds before timing out). This just adds a CLI option to skip it entirely.

Signed-off-by: Yunus Rahbar <yrahbar@fb.com>

pombredanne committed:
Merge branch 'develop' into 2110-dart-pubspec

pombredanne committed:
Merge pull request #2662 from yns88/no-check-version

Add "--no-check-version" CLI option to scancode

pombredanne committed:
Merge pull request #2628 from nexB/2110-dart-pubspec

Align tests for pubspec with latest code

pombredanne committed:
Merge pull request #2625 from akugarg/add_new_licenses

Add new licenses

pombredanne committed:
Merge remote-tracking branch 'itssingh/anti-capitalist' into develop

pombredanne committed:
Remove trailing tab from license data files

Somehow a tab had squeezed in

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not use encoding with json.loads

This is no longer supported in newest Python versions.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne commented:
FYI I merged the latest develop and fixed the tests, then pushed back to your branch

pombredanne reviewed (approved):
LGTM. Merging now at last. Thanks!
